src/main/java/uk/gov/pay/adminusers/service/NewUserExistingServiceInviteCompleter.java

```diff
@@ -49,8 +51,12 @@ public InviteCompleteResponse complete(InviteEntity inviteEntity) {
         if (!inviteEntity.isUserType()) {
             throw internalServerError(format("Attempting to complete a 'new user, existing service' invite for an non user invite. invite-code = %s", inviteEntity.getCode()));
         }
+        if (secondFactor == null) {
+            throw internalServerError(format("Second factor method not provided when attempting to complete a 'new user, existing service' invite. invite-code = %s", inviteEntity.getCode()));
+        }
 
         UserEntity userEntity = inviteEntity.mapToUserEntity();
+        userEntity.setSecondFactor(secondFactor);
```

Comment:
I think it would be better to pass the second factor method into `mapToUserEntity` because otherwise it's set in two places.
It also looks like most of the tests would still pass currently if you removed this line because they are checking the method is set to `SMS`, but we default to `SMS` in `mapToUserEntity`, so you should ensure they fail if we aren't using the `SecondFactorMethod` that comes through in the request.